File: frontend/lib/models/farm.dart
class Farm {
  final int id;
  final String name;
  final String farmer;
  final double latitude;
  final double longitude;
  final String status;
  final double? soilMoisture;
  final double? temperature;
  final String? lastSmsSent;

  Farm({
    required this.id,
    required this.name,
    required this.farmer,
    required this.latitude,
    required this.longitude,
    required this.status,
    this.soilMoisture,
    this.temperature,
    this.lastSmsSent,
  });

  factory Farm.fromJson(Map<String, dynamic> json) {
    return Farm(
      id: json['id'] as int,
      name: json['name'] as String,
      farmer: json['farmer'] as String,
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      status: json['status'] as String,
      soilMoisture: (json['soil_moisture'] as num?)?.toDouble(),
      temperature: (json['temperature'] as num?)?.toDouble(),
      lastSmsSent: json['last_sms_sent'] as String?,
    );
  }
}

File: backend/internal/models/farm.go
package models

type Farm struct {
	ID                int     `json:"id"`
	Name              string  `json:"name"`
	Farmer            string  `json:"farmer"`
	Phone             string  `json:"phone"`
	County            string  `json:"county"`
	SubCounty         string  `json:"sub_county,omitempty"`
	Crop              string  `json:"crop"`
	GrowthStage       string  `json:"growth_stage,omitempty"`
	Latitude          float64 `json:"latitude"`
	Longitude         float64 `json:"longitude"`
	PreferredLanguage string  `json:"preferred_language,omitempty"`
	Status            string  `json:"status"`
	SoilMoisture      float64 `json:"soil_moisture"`
	Temperature       float64 `json:"temperature"`
	RainProbability   float64 `json:"rain_probability"`
	NDVI              float64 `json:"ndvi"`
	WindSpeed         float64 `json:"wind_speed"`
	LastSmsSent       string  `json:"last_sms_sent"`
	CreatedAt         string  `json:"created_at"`
	UpdatedAt         string  `json:"updated_at"`
}
